libevm/VM.cpp

```diff
@@ -296,22 +331,29 @@ void VM::interpretCases()
         CASE(SUICIDE)
         {
             ON_OP();
-            if (m_ext->staticCall)
+            if (m_message->flags & EVM_STATIC)
                 throwDisallowedStateChange();
 
-            m_runGas = toInt63(m_schedule->suicideGas);
-            Address dest = asAddress(m_SP[0]);
+            m_runGas = m_rev >= EVM_TANGERINE_WHISTLE ? 5000 : 0;
+            evm_address destination = toEvmC(asAddress(m_SP[0]));
 
             // After EIP158 zero-value suicides do not have to pay account creation gas.
-            if (m_ext->balance(m_ext->myAddress) > 0 || m_schedule->zeroValueTransferChargesNewAccountGas())
+            evm_uint256be rawBalance;
+            m_context->fn_table->get_balance(&rawBalance, m_context, &m_message->destination);
+            u256 balance = fromEvmC(rawBalance);
+            if (balance > 0 || m_rev < EVM_SPURIOUS_DRAGON)
+            {
                 // After EIP150 hard fork charge additional cost of sending
                 // ethers to non-existing account.
-                if (m_schedule->suicideChargesNewAccountGas() && !m_ext->exists(dest))
-                    m_runGas += m_schedule->callNewAccountGas;
+                int destinationExists =
```

Comment:
Isn't this a bool?

Reply:
No. I use `int` for bools in EVM-C. Bool was added in C99 and looks like `_Bool`.